UPGRADE-3.0.md

```diff
@@ -29,4 +35,76 @@ $recommendation = UserRecommendation::create('user-id', 'scenario');
 $recommendation->setCount(5);
 $recommendation->setRotationRate(1.0);
 $recommendation->setRotationTime(3600);
+```
+
+## `Interaction` now accepts interaction type as parameter
+Matej now allows configuration of custom interaction types and interaction attributes.
+
+At the same time, it allows specifying interaction item using item alias instead
+of item ID. For that reason we removed static constructor methods for creating specific interaction types - `Interaction::detailView`, `Interaction::purchase`, `Interaction::bookmark` and `Interaction::rating`.
+
+We replaced them with constructors for creating Interaction based on `$itemId` or `$itemIdAlias`:
+
+```php
+Interaction::withItem(
+	string $interactionType,
+	string $userId,
+	string $itemId,
+	int $timestamp = null
+```
+
+```php
+Interaction::withAliasedItem(
+	string $interactionType,
+	string $userId,
+	array $itemIdAlias,
+	int $timestamp = null):
+```
+
+The first argument is always a string representing interaction type. Consult the table bellow to find out the correct value to fill in:
+
+| Before: constructor method   | After: argument $interactionType |
+|------------------------------|----------------------------------|
+| `Interaction::detailView`    | `"detailviews"`                  |
+| `Interaction::purchase`      | `"purchases"`                    |
+| `Interaction::bookmark`      | `"bookmarks"`                    |
+| `Interaction::rating`        | `"ratings"`                      |
+
+> To request new interaction types, please contact Matej support.
+
+#### Before
+```php
+$interaction = Interaction::bookmark('user-id', 'item_id', time());
+```
+
+#### After
+```php
+$interaction = Interaction::withItem('bookmarks', 'user-id', 'item_id', time());
+```
+> Argument `$timestamp` remains optional.
+
+## `Interaction` command supports custom attributes
+Interactions now support custom attributes. These can be added using fluent API
+methods `setAttribute()` and `setAttributes()`.
+
+Argument `value` was removed from constructor methods and has to be set using new attribute methods. Its real name might have change as well. For example, for interaction type `bookmarks`, it was renamed to `stars`.
```

Comment:
Is this correct? According to the example below it looks like it was changed for `ratings` interaction.

Reply:
Yes, you I missed that.